yt/frontends/gadget/io.py

```diff
@@ -157,6 +160,10 @@ def _read_particle_fields(self, chunks, ptf, selector):
                     elif field.startswith("Chemistry_"):
                         col = int(field.rsplit("_", 1)[-1])
                         data = g["ChemistryAbundances"][si:ei, col][mask]
+                    elif field == "smoothing_length":
+                        # This is for frontends which don't define a 
+                        # smoothing length
```

Comment:
Can you say something like "smoothing lengths aren't necessarily available on-disk so we don't try to read them", "define a smoothing length" is confusing to me in this context, since it's always well-defined just not necessarily available on-disk.